Flaky test_kraken.py::test_querying_trade_history

## Problem Definition

Happened only once in a travis run https://travis-ci.org/github/rotki/rotki/jobs/699867776

```
kraken = <rotkehlchen.tests.utils.kraken.MockKraken object at 0x7ff68b7a8da0>
    def test_querying_trade_history(kraken):
        now = ts_now()
        result = kraken.query_trade_history(
            start_ts=1451606400,
            end_ts=now,
        )
        assert isinstance(result, list)
>       assert len(result) != 0
E       assert 0 != 0
E        +  where 0 = len([])
rotkehlchen/tests/exchanges/test_kraken.py:63: AssertionError
```

## Task

Try to reproduce and fix it

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -2,6 +2,8 @@
 Changelog
 =========
 
+* :feature: `-` Support Australian Dollar (AUD) as fiat currency
+
 * :release:`1.5.0 <2020-06-10>`
 * :bug: `986` Allows the unsetting of the RPC endpoint
 * :feature: `918` Premium users can now set watchers for their vaults. When the watched vault gets below or above a certain collateralization ratio they get an email alert.

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -3,6 +3,7 @@ Changelog
 =========
 
 * :feature: `-` Support Australian Dollar (AUD) as fiat currency
+* :feature: `-` Count Kraken `off-chain staked assets <https://support.kraken.com/hc/en-us/articles/360039879471-What-is-Asset-S-and-Asset-M->`__ as normal Kraken balance.
 
 * :release:`1.5.0 <2020-06-10>`
 * :bug: `986` Allows the unsetting of the RPC endpoint

--- a/frontend/app/src/data/currencies.ts
+++ b/frontend/app/src/data/currencies.ts
@@ -8,6 +8,7 @@ export const currencies: Currency[] = [
   new Currency('Chinese Yuan', 'CNY', '¥'),
   new Currency('Korean Won', 'KRW', '₩'),
   new Currency('Canadian Dollar', 'CAD', '$'),
+  new Currency('Australian Dollar', 'AUD', '$'),
   new Currency('Russian Ruble', 'RUB', '₽'),
   new Currency('South African Rand', 'ZAR', 'R'),
   new Currency('Turkish Lira', 'TRY', '₺')

--- a/rotkehlchen/assets/asset.py
+++ b/rotkehlchen/assets/asset.py
@@ -68,6 +68,7 @@ WORLD_TO_POLONIEX = {
 WORLD_TO_KRAKEN = {
     'ATOM': 'ATOM',
     'ALGO': 'ALGO',
+    'AUD': 'AUD',
     'BAT': 'BAT',
     'LINK': 'LINK',
     'BSV': 'BSV',

--- a/rotkehlchen/assets/asset.py
+++ b/rotkehlchen/assets/asset.py
@@ -68,7 +68,7 @@ WORLD_TO_POLONIEX = {
 WORLD_TO_KRAKEN = {
     'ATOM': 'ATOM',
     'ALGO': 'ALGO',
-    'AUD': 'AUD',
+    'AUD': 'ZAUD',
     'BAT': 'BAT',
     'LINK': 'LINK',
     'BSV': 'BSV',

--- a/rotkehlchen/assets/converters.py
+++ b/rotkehlchen/assets/converters.py
@@ -453,7 +453,7 @@ def asset_from_kraken(kraken_name: str) -> Asset:
         name = 'BTC'
     elif kraken_name == 'XDG':
         name = 'DOGE'
-    elif kraken_name in ('ETH', 'EUR', 'USD', 'GBP', 'CAD', 'JPY', 'KRW', 'CHF'):
+    elif kraken_name in ('ETH', 'EUR', 'USD', 'GBP', 'CAD', 'JPY', 'KRW', 'CHF', 'AUD'):
         name = kraken_name
     else:
         name = KRAKEN_TO_WORLD.get(kraken_name, kraken_name)

--- a/rotkehlchen/assets/converters.py
+++ b/rotkehlchen/assets/converters.py
@@ -440,7 +440,7 @@ def asset_from_kraken(kraken_name: str) -> Asset:
     if not isinstance(kraken_name, str):
         raise DeserializationError(f'Got non-string type {type(kraken_name)} for kraken asset')
 
-    if kraken_name.endswith('.S'):
+    if kraken_name.endswith('.S') or kraken_name.endswith('.M'):
         # this is a staked coin. For now since we don't show staked coins
         # consider it as the normal version. In the future we may perhaps
         # differentiate between them in the balances

--- a/rotkehlchen/constants/assets.py
+++ b/rotkehlchen/constants/assets.py
@@ -11,7 +11,21 @@ A_RUB = Asset('RUB')
 A_CHF = Asset('CHF')
 A_TRY = Asset('TRY')
 A_ZAR = Asset('ZAR')
-FIAT_CURRENCIES = (A_USD, A_EUR, A_GBP, A_JPY, A_CNY, A_CAD, A_KRW, A_RUB, A_CHF, A_TRY, A_ZAR)
+A_AUD = Asset('AUD')
+FIAT_CURRENCIES = (
+    A_USD,
+    A_EUR,
+    A_GBP,
+    A_JPY,
+    A_CNY,
+    A_CAD,
+    A_KRW,
+    A_RUB,
+    A_CHF,
+    A_TRY,
+    A_ZAR,
+    A_AUD,
+)
 
 S_BTC = 'BTC'
 S_ETH = 'ETH'

--- a/rotkehlchen/data/all_assets.json
+++ b/rotkehlchen/data/all_assets.json
@@ -624,6 +624,11 @@
         "symbol": "AUC",
         "type": "ethereum token"
     },
+    "AUD": {
+        "name": "Australian Dollar",
+        "symbol": "AUD",
+        "type": "fiat"
+    },
     "AUR": {
         "name": "Auroracoin",
         "started": 1393479207,

--- a/rotkehlchen/exchanges/kraken.py
+++ b/rotkehlchen/exchanges/kraken.py
@@ -76,7 +76,7 @@ def kraken_to_world_pair(pair: str) -> TradePair:
     if pair[-2:] == '.d':
         pair = pair[:-2]
 
-    if len(pair) == 6 and pair[0:3] in ('EUR', 'USD'):
+    if len(pair) == 6 and pair[0:3] in ('EUR', 'USD', 'AUD'):
         # This is for the FIAT to FIAT pairs that kraken introduced
         base_asset_str = pair[0:3]
         quote_asset_str = pair[3:]

--- a/rotkehlchen/tests/exchanges/test_kraken.py
+++ b/rotkehlchen/tests/exchanges/test_kraken.py
@@ -27,6 +27,9 @@ def test_coverage_of_kraken_balances(kraken):
     expected_assets = (set(KRAKEN_TO_WORLD.keys()) - set(KRAKEN_DELISTED)).union({'BSV'})
     # Ignore the staking assets from the got assets
     got_assets.remove('XTZ.S')
+    got_assets.remove('EUR.M')
+    got_assets.remove('USD.M')
+    got_assets.remove('XBT.M')
 
     diff = expected_assets.symmetric_difference(got_assets)
     if len(diff) != 0:
@@ -41,6 +44,7 @@ def test_coverage_of_kraken_balances(kraken):
 
     # also check that staked assets are properly processed
     assert asset_from_kraken('XTZ.S') == Asset('XTZ')
+    assert asset_from_kraken('EUR.M') == Asset('EUR')
 
 
 def test_querying_balances(function_scope_kraken):

--- a/rotkehlchen/tests/exchanges/test_kraken.py
+++ b/rotkehlchen/tests/exchanges/test_kraken.py
@@ -43,8 +43,8 @@ def test_coverage_of_kraken_balances(kraken):
     assert asset_from_kraken('XTZ.S') == Asset('XTZ')
 
 
-def test_querying_balances(kraken):
-    result, error_or_empty = kraken.query_balances()
+def test_querying_balances(function_scope_kraken):
+    result, error_or_empty = function_scope_kraken.query_balances()
     assert error_or_empty == ''
     assert isinstance(result, dict)
     for asset, entry in result.items():
@@ -53,9 +53,9 @@ def test_querying_balances(kraken):
         assert 'amount' in entry
 
 
-def test_querying_trade_history(kraken):
+def test_querying_trade_history(function_scope_kraken):
     now = ts_now()
-    result = kraken.query_trade_history(
+    result = function_scope_kraken.query_trade_history(
         start_ts=1451606400,
         end_ts=now,
     )
@@ -66,9 +66,9 @@ def test_querying_trade_history(kraken):
         assert isinstance(kraken_trade, Trade)
 
 
-def test_querying_deposits_withdrawals(kraken):
+def test_querying_deposits_withdrawals(function_scope_kraken):
     now = ts_now()
-    result = kraken.query_deposits_withdrawals(
+    result = function_scope_kraken.query_deposits_withdrawals(
         start_ts=1451606400,
         end_ts=now,
     )

--- a/rotkehlchen/tests/exchanges/test_kraken.py
+++ b/rotkehlchen/tests/exchanges/test_kraken.py
@@ -101,6 +101,7 @@ def test_kraken_to_world_pair(kraken):
     assert kraken_to_world_pair('LTCUSDT') == 'LTC_USDT'
     assert kraken_to_world_pair('XRPGBP') == 'XRP_GBP'
     assert kraken_to_world_pair('XRPUSDT') == 'XRP_USDT'
+    assert kraken_to_world_pair('AUDJPY') == 'AUD_JPY'
 
     # now try to test all pairs that kraken returns and if one does not work note
     # down a test warning so that it can be fixed by us later

--- a/rotkehlchen/tests/utils/kraken.py
+++ b/rotkehlchen/tests/utils/kraken.py
@@ -270,7 +270,7 @@ def generate_random_single_kraken_ledger_data(
     return ledger
 
 
-def generate_random_kraken_ledger_data(start: Timestamp, end: Timestamp, ledger_type):
+def generate_random_kraken_ledger_data(start: Timestamp, end: Timestamp, ledger_type: str):
     ledgers_num = random.randint(1, 49)
     # Ledgers is a dict with txid as the key
     ledgers = {}

